src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10490,6 +10490,282 @@ public void SynthesizedDelegateTypes_NamedArguments_MoreThan16Parameters()
+                Report(lam2);
+                var lam3 = (string c = "abc") => { };
+                Report(lam3);
+                var lam4 = (string d = "a" + "bc") => { };
+                Report(lam4);
+                var lam5 = ([Optional, DefaultParameterValue("abc")] object o) => { };
+                Report(lam5);
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`1[System.String]
+                <>f__AnonymousDelegate1`1[System.String]
+                <>f__AnonymousDelegate2`1[System.String]
+                <>f__AnonymousDelegate2`1[System.String]
+                <>f__AnonymousDelegate2`1[System.Object]
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_ParamsArray_DifferentElementTypes()
+        {
+            var source = """
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = (params int[] xs) => { };
+                Report(lam1);
+                var lam2 = (params string[] ys) => { };
+                Report(lam2);
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`1[System.Int32]
+                <>f__AnonymousDelegate0`1[System.String]
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_ParamsArray_NonArrayTypeInMetadata()
+        {
+            /*
+                public static class C
+                {
+                    public static void M(params int x) { }
+                }
+             */
+            var ilSource = """
+                .class public auto ansi abstract sealed beforefieldinit C
+                    extends [mscorlib]System.Object
+                {
+                    // Methods
+                    .method public hidebysig static 
+                        void M (
+                            int32 x
+                        ) cil managed 
+                    {
+                        .param [1]
+                            .custom instance void [mscorlib]System.ParamArrayAttribute::.ctor() = (
+                                01 00 00 00
+                            )
+                        // Method begins at RVA 0x20a2
+                        // Code size 2 (0x2)
+                        .maxstack 8
+
+                        IL_0000: nop
+                        IL_0001: ret
+                    } // end of method C::M
+
+                } // end of class C
+                """;
+            var source = """
+                var m = C.M;
+                System.Console.WriteLine(m.GetType());
+                """;
+            var comp = CreateCompilationWithIL(source, ilSource).VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "System.Action`1[System.Int32]");
```

Comment:
>CompileAndVerify [](http://example.com/codeflow?start=12&length=16)

Consider always calling  ```VerifyDiagnostics``` on result of ```CompileAndVerify```, unless diagnostics was checked beforehand. Are we going to get warning about `params' mismatch?